api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -96,18 +98,51 @@ func (m *AzureManagedControlPlane) setDefaultSubnet() {
 	}
 }
 
-func setDefaultSku(sku *AKSSku) *AKSSku {
-	if sku == nil {
-		return &AKSSku{
-			Tier: FreeManagedControlPlaneTier,
+func setDefaultSku(sku *AKSSku) {
+	if sku != nil && sku.Tier == PaidManagedControlPlaneTier {
+		sku.Tier = StandardManagedControlPlaneTier
+		ctrl.Log.WithName("AzureManagedControlPlaneWebHookLogger").Info("Paid SKU tier is deprecated and has been replaced by Standard")
+	}
+}
+
+func setDefaultNetworkPlugin(networkPlugin *string) {
+	if networkPlugin == nil {
+		networkPlugin = ptr.To(AzureNetworkPluginName)
```

Comment:
I think the linter is saying that for this to have the desired side effect, this would have to do `*networkPlugin = ...` instead of `networkPlugin = ...`. The reason the SKU one above doesn't throw the same error is because there only a field on a struct pointer is set, so the value of the input pointer isn't changed like it is here.

All in all I think it would be this:
```suggestion
		networkPlugin = new(string)
		*networkPlugin = AzureNetworkPluginName
```

Are there any tests that also catch this?

Reply:
This seems to not work with the field on a struct pointer such as `setDefaultSku`, even when using this method. I'm wondering if there is a better way to do this... Also I think we can just use kubebuilder the default here.